plugin/storage/integration/integration.go

```diff
@@ -481,6 +507,7 @@ func (s *StorageIntegration) insertThroughput(t *testing.T) {
 // IntegrationTestAll runs all integration tests
 func (s *StorageIntegration) IntegrationTestAll(t *testing.T) {
 	t.Run("GetServices", s.testGetServices)
+	t.Run("ArchiveTrace", s.testArchiveTrace)
```

Comment:
so there's nothing in this function that makes it "archive" - it's just using the main writer/reader. In some sense this is not bad conceptually (I even have another ticket to deprecate "archive" APIs), but when the storage has specific behavior only for archive mode your test is not going to exercise that behavior. Specifically, although the `testArchiveTrace` in ES looked like it also uses primary reader/writer, those were actually initialized with an archive=true flag (see L211 in plugin/storage/es/spanstore/reader.go), unlike the rest of the tests.

Reply:
Ok, thanks a lot for the clarification. I went through the ES Test and did notice the archive bool value initializing ES. I just had one more query, what about the other backends like Cassandra, Badger etc. Because, they aren't initialized in the same manner. They aren't initialized with the archive bool value, so how do I get them to exercise the archive mode behaviour?

Reply:
Right now the shared tests run by assuming that SpanReader etc. fields in the struct are set, and it's up to individual storage implementations to initialize those. What we can do is add ArchiveSpanReader/etc to the main struct and have the implementations initialize them as they wish. This way ES can create separate reader/write with archive=true, while other implementations can just use the same reader/writer since they don't have functional differences.

Reply:
Ok got it, I did initialize Archive Span Reader/Writer i the main struct earlier, but writing the test required initialization of a storage factory, will revert back to the change and work on it. Thanks!